app/models/canonical/sync/syncer.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Canonical
+  module Sync
+    class Syncer
+      def self.perform(preserve_existing_records)
+        new(preserve_existing_records).perform
+      end
+
+      def initialize(preserve_existing_records)
+        @preserve_existing_records = preserve_existing_records
+      end
+
+      def perform
+        Rails.logger.info "Syncing #{model_name.downcase.pluralize}..."
+
+        ActiveRecord::Base.transaction do
+          destroy_existing_models unless preserve_existing_records
+
+          json_data.each {|object| create_or_update_model(object[:attributes]) }
+        rescue StandardError => e
+          Rails.logger.error "Unexpected error #{e.class} while syncing #{model_name.downcase.pluralize}: #{e.message}"
+          raise e
+        end
+      end
+
+      private
+
+      attr_reader :preserve_existing_records
+
+      def model_class
+        raise NotImplementedError.new('Child class of Canonical::Sync::Syncer must define a model class')
+      end
+
+      def model_name
+        model_class.to_s.scan(/[A-Z][a-z]*/).join(' ')
```

Comment:
This will admittedly break if there is a name like `HTMLParser` that contains an abbreviation, but that isn't really anticipated in this code base.